logging/config/config.go

```diff
@@ -0,0 +1,80 @@
+package config
+
+import (
+	"bytes"
+
+	"fmt"
+
+	"github.com/BurntSushi/toml"
+	"github.com/eris-ltd/eris-db/logging/structure"
+)
+
+type LoggingConfig struct {
+	InfoSink         *SinkConfig `toml:"info_sink"`
+	InfoAndTraceSink *SinkConfig `toml:"info_and_trace_sink"`
+}
+
+// For encoding a top-level '[logging]' TOML table
+type LoggingConfigWrapper struct {
+	Logging *LoggingConfig `toml:"logging"`
+}
+
+func DefaultNodeLoggingConfig() *LoggingConfig {
+	return &LoggingConfig{
+		InfoSink:         Sink(),
+		InfoAndTraceSink: Sink().SetOutput(StderrOutput()),
+	}
+}
+
+func DefaultClientLoggingConfig() *LoggingConfig {
+	return &LoggingConfig{
+		// No output
+		InfoSink: Sink().
+			SetTransform(FilterTransform(ExcludeWhenAnyMatches,
+				structure.CapturedLoggingSourceKey, "")).
+			SetOutput(StderrOutput()),
+		InfoAndTraceSink: Sink(),
+	}
+}
+
+// Returns the TOML for a top-level logging config wrapped with [logging]
+func (lc *LoggingConfig) RootTOMLString() string {
+	return tomlString(LoggingConfigWrapper{lc})
+}
+
+func (lc *LoggingConfig) TOMLString() string {
+	return tomlString(lc)
+}
+
+func LoggingConfigFromMap(loggingRootMap map[string]interface{}) (*LoggingConfig, error) {
+	lc := new(LoggingConfig)
+	buf := new(bytes.Buffer)
+	enc := toml.NewEncoder(buf)
+	// TODO: [Silas] consider using strongly typed config/struct mapping everywhere
+	// (!! unfortunately the way we are using viper
+	// to pass around a untyped bag of config means that we don't get keys mapped
+	// according to their metadata `toml:"Name"` tags. So we are re-encoding to toml
+	// and then decoding into the strongly type struct as a work-around)
+	// Encode the map back to TOML
+	err := enc.Encode(loggingRootMap)
+	if err != nil {
+		return nil, err
+	}
+	// Decode into struct into the LoggingConfig struct
+	_, err = toml.Decode(buf.String(), lc)
+	if err != nil {
+		return nil, err
+	}
+	return lc, nil
+}
+
+func tomlString(v interface{}) string {
+	buf := new(bytes.Buffer)
+	encoder := toml.NewEncoder(buf)
+	err := encoder.Encode(v)
+	if err != nil {
+		// Seems like a reasonable compromise to make the string function clean
```

Comment:
The reasoning is, this will generate pretty obviously failing config if it does slip through, shouldn't cause a runtime issue, and primarily because failures are highly likely to be caught by tests. It doesn't seem reasonable to error or panic on encoding a well-known type like this.

That said, I also feel borderline about this.